lib/sockets-monitor.js

```diff
@@ -0,0 +1,59 @@
+'use strict';
+var child = require('child_process');
+
+
+module.exports = SocketsMonitor;
+
+/**
+ * Monitor class for performing monitoring on:
+ * - number of **open** sockets
+ * - number of **pending** sockets
+ * - number of **open** files
+ * @class
+ */
+function SocketsMonitor (monitor, prefix, interval) {
+  this.monitor = monitor;
+  this.prefix = prefix;
+  this.interval = interval;
+}
+
+/**
+ * Start capturing of the sockets stats data.
+ */
+SocketsMonitor.prototype.start = function () {
+  if (this.intervalId) { return; }
+  this.intervalId = setInterval(this._captureSocketStats.bind(this), this.interval);
+};
+
+/**
+ * Stop capturing of the sockets stats data.
+ */
+SocketsMonitor.prototype.stop = function () {
+  if (this.intervalId) {
+    clearInterval(this.intervalId);
+    this.intervalId = null;
+  }
+};
+
+SocketsMonitor.prototype._captureSocketStats = function () {
+  var sockets = require('http').globalAgent.sockets;
+  var request = require('http').globalAgent.requests;
+  var key;
```

Comment:
Do we use `key` outside the scope of the for loops below? If not, should we just locally scope the variable there? (i.e. `for (var key in ...)`)